CefSharp.Core/Internals/CefURLRequestClientAdapter.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright © 2019 The CefSharp Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+#include "Stdafx.h"
+#include "CefURLRequestClientAdapter.h"
+#include "CefURLRequestWrapper.h"
+#include "CefAuthCallbackWrapper.h"
+
+#include "Cef.h"
+
+
+void CefURLRequestClientAdapter::OnRequestComplete(CefRefPtr<CefURLRequest> request)
+{
+    _client->OnRequestComplete(gcnew CefURLRequestWrapper(request));
+}
+
+
+void CefURLRequestClientAdapter::OnUploadProgress(CefRefPtr<CefURLRequest> request, int64 current, int64 total)
+{
+    _client->OnUploadProgress(gcnew CefURLRequestWrapper(request), current, total);
+}
+
+void CefURLRequestClientAdapter::OnDownloadProgress(CefRefPtr<CefURLRequest> request, int64 current, int64 total)
+{
+    _client->OnDownloadProgress(gcnew CefURLRequestWrapper(request), current, total);
+}
+
+void CefURLRequestClientAdapter::OnDownloadData(CefRefPtr<CefURLRequest> request, const void* data, size_t data_length)
+{
+    auto data_array = gcnew array<Byte>(data_length);
```

Comment:
Use `UnmanagedMemoryStream` to wrap the data in a `Stream.

Reply:
Done -- could you please make sure my memory management here is correct? I expect it should get deallocated once we leave the stack frame, but I'm not super familiar with how C++\CLI works.